src/compiler/checker.ts

```diff
@@ -29800,17 +29801,15 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             !!(type.flags & TypeFlags.Instantiable && !maybeTypeOfKind(getBaseConstraintOrType(type), TypeFlags.Nullable));
     }
 
-    function hasContextualTypeWithNoGenericTypes(node: Node, checkMode: CheckMode | undefined) {
+    function hasContextualTypeWithNoGenericTypes(node: Node, checkMode = CheckMode.Normal) {
         // Computing the contextual type for a child of a JSX element involves resolving the type of the
         // element's tag name, so we exclude that here to avoid circularities.
         // If check mode has `CheckMode.RestBindingElement`, we skip binding pattern contextual types,
         // as we want the type of a rest element to be generic when possible.
         const contextualType = (isIdentifier(node) || isPropertyAccessExpression(node) || isElementAccessExpression(node)) &&
             !((isJsxOpeningElement(node.parent) || isJsxSelfClosingElement(node.parent)) && node.parent.tagName === node) &&
-            (checkMode && checkMode & CheckMode.RestBindingElement ?
-                getContextualType(node, ContextFlags.SkipBindingPatterns)
-                : getContextualType(node, /*contextFlags*/ undefined));
-        return contextualType && !isGenericType(contextualType);
+            getContextualType(node, (checkMode & CheckMode.RestBindingElement ? ContextFlags.SkipBindingPatterns : 0) | (checkMode & CheckMode.SatisfiesOutput ? ContextFlags.SkipSatisfies : 0));
+        return contextualType && !(contextualType.flags & TypeFlags.Any) && !isGenericType(contextualType);

```

Comment:
The added `!(contextualType.flags & TypeFlags.Any)` bit here isn't exactly related to `satisfies` at all. I have added it to account for auto types, that affected all the baselines that had `any` as the contextual type in a few places. It also affected those codefixes for missing function declarations since the error type (a special kind of any) was the contextual type for those arguments here. I think those changes are positive